### PR TITLE
Support OptionalType to get shape | feat(graph_building)

### PR DESCRIPTION
**Pull request by titaiwangms**
Due to the need of supporting None args, we need to support None (torch.OptionalType) to get shape/dtype.
To support https://github.com/pytorch/pytorch/pull/105263

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]
Possible to add a test?

File: onnxscript/function_libs/torch_lib/graph_building_test.py [hunk 1]
```diff
@@ -118,6 +118,12 @@ def outer(x: FLOAT[1, 2, 3]):
         expected = outer.to_model_proto()
         onnxscript.testing.assert_isomorphic(traced, expected)
 
+    def test_add_input_works_with_optionaltype(self):
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/graph_building_test.py:121]: Would be nice to clarify what "works with" mean? Does it mean None can now be an input to the graph, or something else?

Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/graph_building_test.py]: How about test_add_input_with_optionaltype_does_not_raise_torch_internal_error

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/graph_building_test.py]: SG!

**Pull request merged**